core/src/edu/cornell/gdiac/main/GameController.java

```diff
@@ -159,50 +223,49 @@ public GameController() {
    *
    * @return true if the level is completed.
    */
-  public boolean isComplete() {
-    return complete;
+  public boolean isGameComplete() {
+    return checkCompleteState(this.gameState);
   }
 
+
   /**
-   * Sets whether the level is completed.
-   * <p>
-   * If true, the level will advance after a countdown
+   * Sets the game state of this controller
    *
-   * @param value whether the level is completed.
+   * @param nextGameState The state to which to transition this controller
    */
-  public void setComplete(boolean value) {
-    if (value) {
+  public void setGameState(GameState nextGameState) {
+    gameState = nextGameState;
+
+    boolean isNextComplete = checkCompleteState(nextGameState);
+    if (isNextComplete) {
+      // Set countdown for whether to passively transition to next level after some time
       countdown = EXIT_COUNT;
     }
-    complete = value;
-
-    level.setComplete(value);
-  }
 
-  /**
-   * Returns true if the level is failed.
-   * <p>
-   * If true, the level will reset after a countdown
-   *
-   * @return true if the level is failed.
-   */
-  public boolean isFailure() {
-    return failed;
+    // NOTE: level state can probably be simplified to match how level completion is stored in this
+    // Update level state
+    level.setComplete(isNextComplete);
+    level.setFailure(gameState == GameState.LEVEL_FAILURE);
+
+    // Menu display logic
+    hideAllMenus();
+    if (screenActive) {
+      if (nextGameState == GameState.LEVEL_SUCCESS) {
+        successMenu.show();
+      } else if (nextGameState == GameState.LEVEL_FAILURE) {
+        failureMenu.show();
+      } else if (nextGameState == GameState.PAUSED) {
```

Comment:
~~I noticed that you do specify a "pause" state, but nothing about the game actually pauses when we are in this state. This is fine for now I guess, but I am definitely going to have someone implement that before beta.~~

Edit: I didn't realize that you activate pause with `x`! I think it would make more sense to map pause to `Esc` and remove the level selector key-bind since it can be accessed by the pause menu anyway. This would make it more intuitive for testers of our game.

Reply:
Sounds good